src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -361,6 +366,264 @@ namespace cryptonote
+      // moneromooo ... only because I <3 him. 
+      std::list<size_t> need_tx_indices;
+        
+      transaction tx;
+      crypto::hash tx_hash;
+      std::unordered_map<crypto::hash, blobdata> tx_hash_to_blob;        
+      BOOST_FOREACH(auto& tx_blob, arg.b.txs)
+      {
+        if(parse_and_validate_tx_from_blob(tx_blob, tx))
+        {
+          try
+          {
+            if(!get_transaction_hash(tx, tx_hash))
+            {
+              LOG_PRINT_CCONTEXT_L1("get_transaction_hash failed");
+              m_p2p->drop_connection(context);
+              m_core.resume_mine();
+              return 1;
+            }
+          }
+          catch(...)
+          {
+            LOG_PRINT_CCONTEXT_L1("get_transaction_hash failed");
+            m_p2p->drop_connection(context);
+            m_core.resume_mine();
+            return 1;
+          }
+          
+          // we might already have the tx that the peer
+          // sent in our pool, so don't verify again..
+          if(!m_core.get_pool_transaction(tx_hash, tx))
+          {
+            cryptonote::tx_verification_context tvc = AUTO_VAL_INIT(tvc);                        
+            if(!m_core.handle_incoming_tx(tx_blob, tvc, true, true) || tvc.m_verifivation_failed)
+            {
+              LOG_PRINT_CCONTEXT_L1("Block verification failed: transaction verification failed, dropping connection");
+              m_p2p->drop_connection(context);
+              m_core.resume_mine();
+              return 1;
+            }
+            
+            // Should only not be added to pool if verification failed, but
+            // maybe in the future could not be added for other reasons 
+            // according to monero-moo so keep track of these separately ..
+            //
+            // this is probably the wrong thing to do, but for now the idea
+            // would be to pass these back/forth on the missing tx request,
+            // right now they're just being lost at the end of this function call
+            if(!tvc.m_added_to_pool) 
+              tx_hash_to_blob[tx_hash] = tx_blob;
+          }
+        }
+        else
+        {
+          LOG_ERROR_CCONTEXT
+          (
+            "sent wrong tx: failed to parse and validate transaction: \r\n"
+            << epee::string_tools::buff_to_hex_nodelimer(tx_blob) 
+            << "\r\n dropping connection"
+          );
+            
+          m_p2p->drop_connection(context);
+          m_core.resume_mine();
+          return 1;
+        }
+      }
+      
+      size_t tx_idx = 0;
+      BOOST_FOREACH(auto& tx_hash, new_block.tx_hashes)
+      {
+        if(m_core.get_pool_transaction(tx_hash, tx))
+        {
+            have_tx.push_back(tx_to_blob(tx));
+        }
+        else
+        {
+          if(tx_hash_to_blob.find(tx_hash) != tx_hash_to_blob.end()) 
+          {
+              have_tx.push_back(tx_hash_to_blob[tx_hash]);
+          }
```

Comment:
This is still wrong, since the next fluffy block call will not have that tx, so you will end up in a ping pong that never ends.